pkg/manager/leaderelection.go

```diff
@@ -3,28 +3,86 @@ package manager
 import (
 	"context"
 
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/k8snetworkplumbingwg/kubemacpool/pkg/names"
 )
 
 func (k *KubeMacPoolManager) waitToStartLeading() error {
 	<-k.mgr.Elected()
-	return k.markPodAsLeader()
+	err := k.setLeadershipConditions(corev1.ConditionFalse)
```

Comment:
did you mean to add this before the <-k.mgr.Elected() so that both pods will be set to false ( until the elected one is chosen)?

Reply:
Nope, If we set it before it just cover the startup case not the ledership change.

Reply:
so, why set if to OFF only to set it to on again?